File: javascript/src/api/background_task.ts
// this file is @generated
import { BackgroundTaskOut, BackgroundTaskOutUtil } from "../models/background_task_out";
import {
  BackgroundTaskStatus,
  BackgroundTaskStatusUtil,
} from "../models/background_task_status";
import {
  BackgroundTaskType,
  BackgroundTaskTypeUtil,
} from "../models/background_task_type";
import {
  ListResponseBackgroundTaskOut,
  ListResponseBackgroundTaskOutUtil,
} from "../models/list_response_background_task_out";
import { Ordering, OrderingUtil } from "../models/ordering";
import { HttpMethod, SvixRequest, SvixRequestContext } from "../request";

export interface BackgroundTaskListOptions {
  /** Filter the response based on the status. */
  status?: BackgroundTaskStatus;
  /** Filter the response based on the type. */
  task?: BackgroundTaskType;
  /** Limit the number of returned items */
  limit?: number;
  /** The iterator returned from a prior invocation */
  iterator?: string | null;
  /** The sorting order of the returned items */
  order?: Ordering;
}

export class BackgroundTask {
  public constructor(private readonly requestCtx: SvixRequestContext) {}

  /** List background tasks executed in the past 90 days. */
  public list(
    options?: BackgroundTaskListOptions
  ): Promise<ListResponseBackgroundTaskOut> {
    const request = new SvixRequest(HttpMethod.GET, "/api/v1/background-task");

    request.setQueryParam("status", options?.status);
    request.setQueryParam("task", options?.task);
    request.setQueryParam("limit", options?.limit);
    request.setQueryParam("iterator", options?.iterator);
    request.setQueryParam("order", options?.order);

    return request.send(
      this.requestCtx,
      ListResponseBackgroundTaskOutUtil._fromJsonObject
    );
  }

  /**
   * List background tasks executed in the past 90 days.
   *
   * @deprecated Use list instead.
   * */
  public listByEndpoint(
    options?: BackgroundTaskListOptions
  ): Promise<ListResponseBackgroundTaskOut> {
    return this.list(options);
  }

  /** Get a background task by ID. */
  public get(taskId: string): Promise<BackgroundTaskOut> {
    const request = new SvixRequest(HttpMethod.GET, "/api/v1/background-task/{task_id}");

    request.setPathParam("task_id", taskId);

    return request.send(this.requestCtx, BackgroundTaskOutUtil._fromJsonObject);
  }
}

File: kotlin/lib/src/main/kotlin/BackgroundTask.kt
// this file is @generated
package com.svix.kotlin

import com.svix.kotlin.models.BackgroundTaskOut
import com.svix.kotlin.models.BackgroundTaskStatus
import com.svix.kotlin.models.BackgroundTaskType
import com.svix.kotlin.models.ListResponseBackgroundTaskOut
import com.svix.kotlin.models.Ordering
import okhttp3.HttpUrl

data class BackgroundTaskListOptions(
    val status: BackgroundTaskStatus? = null,
    val task: BackgroundTaskType? = null,
    val limit: ULong? = null,
    val iterator: String? = null,
    val order: Ordering? = null,
)

class BackgroundTask(baseUrl: HttpUrl, defaultHeaders: Map<String, String>) :
    SvixHttpClient(baseUrl, defaultHeaders) {

    /** List background tasks executed in the past 90 days. */
    suspend fun list(
        options: BackgroundTaskListOptions = BackgroundTaskListOptions()
    ): ListResponseBackgroundTaskOut {
        var url = this.newUrlBuilder().encodedPath("/api/v1/background-task")
        options.status?.let { url = url.addQueryParameter("status", it.toString()) }
        options.task?.let { url = url.addQueryParameter("task", it.toString()) }
        options.limit?.let { url = url.addQueryParameter("limit", it.toString()) }
        options.iterator?.let { url = url.addQueryParameter("iterator", it) }
        options.order?.let { url = url.addQueryParameter("order", it.toString()) }
        return this.executeRequest<Any, ListResponseBackgroundTaskOut>("GET", url.build())
    }

    /** Get a background task by ID. */
    suspend fun get(taskId: String): BackgroundTaskOut {
        val url = this.newUrlBuilder().encodedPath("/api/v1/background-task/$taskId")
        return this.executeRequest<Any, BackgroundTaskOut>("GET", url.build())
    }
}
